src/core/shape_inference/include/nms_shape_inference.hpp

```diff
@@ -60,16 +65,24 @@ void shape_infer(const NonMaxSuppression* op, const std::vector<T>& input_shapes
     // NonMaxSuppression produces triplets
     // that have the following format: [batch_index, class_index, box_index]
     ov::PartialShape out_shape = {Dimension::dynamic(), 3};
-    if (boxes_ps.rank().is_static() && scores_ps.rank().is_static() && op->get_input_size() > 2) {
+    if (boxes_ps.rank().is_static() && scores_ps.rank().is_static()) {
         const auto num_boxes_boxes = boxes_ps[1];
-        if (num_boxes_boxes.is_static() && scores_ps[0].is_static() && scores_ps[1].is_static() &&
-            has_and_set_equal_bounds(op->input_value(2))) {
+        if (num_boxes_boxes.is_static() && scores_ps[0].is_static() && scores_ps[1].is_static()) {
             const auto num_boxes = num_boxes_boxes.get_length();
             const auto num_classes = scores_ps[1].get_length();
-            const auto max_output_boxes_per_class = op->max_boxes_output_from_input();
-
-            out_shape[0] =
-                Dimension(0, std::min(num_boxes, max_output_boxes_per_class) * num_classes * scores_ps[0].get_length());
+            int64_t max_output_boxes_per_class = 0;
+            std::vector<int64_t> max_output_boxes_per_class_as_vals;
+            bool status = get_data_as_int64<T>(1, op, max_output_boxes_per_class_as_vals, constant_data);
+            if (constant_data.count(1) && status) {
+                max_output_boxes_per_class = max_output_boxes_per_class_as_vals[0];
+                out_shape[0] =
+                    std::min(num_boxes, max_output_boxes_per_class) * num_classes * scores_ps[0].get_length();
+            } else if (op->get_input_size() > 2 && has_and_set_equal_bounds(op->input_value(2))) {
+                max_output_boxes_per_class = op->max_boxes_output_from_input();
+                out_shape[0] = Dimension(
+                    0,
+                    std::min(num_boxes, max_output_boxes_per_class) * num_classes * scores_ps[0].get_length());
+            }
```

Comment:
```suggestion
            std::vector<int64_t> max_output_boxes_per_class_as_vals;
            bool data_available = get_data_as_int64<T>(1, op, max_output_boxes_per_class_as_vals, constant_data) || (op->get_input_size() > 2 && get_data_as_int64<T>(2, op, max_output_boxes_per_class_as_vals, constant_data));
            if (data_available) {
                int64_t max_output_boxes_per_class = max_output_boxes_per_class_as_vals[0];
                out_shape[0] =
                    std::min(num_boxes, max_output_boxes_per_class) * num_classes * scores_ps[0].get_length();
            }
```

Reply:
@jane-intel I found that dynamic output shape is needed in case of existing shape infer like ngraph reference, whereas the static shape is required in shape infer of GPU plugin. In this case, how do I combine them(if+elseif)?

Reply:
For existing case that shape infer uses inputs w/o constant data still requires a dynamic output shape, so `static_output` default parameter has been added.